packages/protocol/contracts/L1/TaikoToken.sol

```diff
@@ -84,4 +84,8 @@ contract TaikoToken is EssentialContract_, ERC20VotesUpgradeable {
         if (_to == address(this)) revert TKO_INVALID_ADDR();
         return super.transferFrom(_from, _to, _amount);
     }
+
+    function clock() public view override returns (uint48) {
+        return SafeCastUpgradeable.toUint48(block.timestamp);
```

Comment:
shall I use `SafeCastUpgradeable.toUint48(block.timestamp/ SOME_CONSTANT)` instead?

Reply:
We needCLOCK_MODE() override too:
Something like:

```
// solhint-disable-next-line func-name-mixedcase 
function CLOCK_MODE() public pure override returns (string memory) { 
    return "mode=timestamp"; 
}
```

Reply:
@dantaik I would suggst keeping the best practices from OpenZeppelin and keep the consistency check: 
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/utils/Votes.sol#L68-L71

Reply:
Thank you, Jordi, but since the clock() function is no longer a virtual function, I think removing the check is fine.